semanticscholar/ApiRequester.py

```diff
@@ -59,9 +56,13 @@ def get_data(
 
         url = f'{url}?{parameters}'
         method = 'POST' if payload else 'GET'
-        payload = json.dumps(payload) if payload else None
-        r = requests.request(
-            method, url, timeout=self._timeout, headers=headers, data=payload)
+
+        async with httpx.AsyncClient() as client:
+            try:
+                r = await client.request(
+                    method, url, timeout=self._timeout, headers=headers, json=payload)
+            except httpx.TimeoutException:
+                raise TimeoutException("Request timed out.")
```

Comment:
I don't see the need to catch the exception here, since we do nothing more than inform that a Timeout exception occurred.